fix(bundler): Use `arch` instead of `llvm_target`. fix #3285

JonasKruckenberg opened the pull request:
The llvm target misidentified the arch on apple silicon, causing sidecar bundling to fail.
This fixes #3285.

<!--
Update "[ ]" to "[x]" to check a box

Please make sure to read the Pull Request Guidelines: https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information

JonasKruckenberg committed:
fix(bundler): Use `arch` instead of `llvm_target`

The llvm target misidentified the arch on apple silicon, causing sidecar bundling to fail.
This fixes #3285.

lucasfernog commented:
I see you didn't change the command to `rustc --print cfg` as stated on the issue. Is that intentional?

JonasKruckenberg commented:
> I see you didn't change the command to `rustc --print cfg` as stated on the issue. Is that intentional?

nope, just didn't get to it yet bc the parsing is a bit more involved

JonasKruckenberg committed:
fix: Use `rustc --print cfg` for target arch

Replaces the nightly (and undocumented?) `RUSTC_BOOTSTRAP=1 rustc -Z unstable-options --print target-spec-json` with `rustc --print cfg` which is on the public interface, documented and fully uses stable rust

JonasKruckenberg committed:
add changeset

lucasfernog committed:
Merge remote-tracking branch 'origin/next' into fix/incorrect-llvm-target

lucasfernog committed:
chore: do not use regex

lucasfernog committed:
fix: clippy
